errored.go

```diff
@@ -152,11 +152,13 @@ func (e *Error) Error() string {
 	}
 
 	if e.trace || AlwaysTrace {
-		ret := desc + "\n"
-
-		for _, stack := range e.stack {
-			for _, line := range stack {
-				ret += fmt.Sprintf("\t%s [%s %d]\n", line.fun, line.file, line.line)
+		var ret string
+		for i := 0; i < len(e.errors); i++ {
+			ret += fmt.Sprintf("%s\n", e.errors[i].Error())
+			if len(e.stack) > i {
```

Comment:
@yuva29 , I think we only keep stack for errors of type `errored.Error`

@erikh 
I might be missing something here, but I think if we were to combine a mix of errored and non-errored errors then the stack trace association might not come out correct in some cases:
```go
func func1 () error {
    return errored.New("foobar")
}

func func1Cleanup() error {
    return fmt.Errorf("bar")
}

func func2 () error {
   if e1 = func1(); e1 != nil {
      if e2 = func1Cleanup(); e2 != nil  {
            return errored.New("foo").Combine(e2).Combine(e1)  // the stack for this resulting error will mismatch the stacked errors
      }
      return errored.New("foo").Combine(e1)
   }
}
...
...
```